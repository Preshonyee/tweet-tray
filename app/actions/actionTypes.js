export const UPDATE_ACCESS_TOKEN_PAIR = 'UPDATE_ACCESS_TOKEN_PAIR';
export const UPDATE_REQUEST_TOKEN_PAIR = 'UPDATE_REQUEST_TOKEN_PAIR';

export const SET_PROFILE_IMAGE_URL = 'SET_PROFILE_IMAGE_URL';
export const SET_PROFILE_LINK_COLOR = 'SET_PROFILE_LINK_COLOR';
export const UPDATE_WEIGHTED_STATUS = 'UPDATE_WEIGHTED_STATUS';

export const TOGGLE_LAUNCH_ON_START_UP = 'TOGGLE_LAUNCH_ON_START_UP';
export const TOGGLE_COLOR_THEME = 'TOGGLE_COLOR_THEME';

export const ON_LOGOUT = 'ON_LOGOUT';
