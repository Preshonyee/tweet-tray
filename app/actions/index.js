import * as types from './actionTypes';

export const updateAccessTokenPair = (accessTokenPair) => {
  return {
    type: types.UPDATE_ACCESS_TOKEN_PAIR,
    accessTokenPair,
  };
};

export const updateRequestTokenPair = (requestTokenPair) => {
  return {
    type: types.UPDATE_REQUEST_TOKEN_PAIR,
    requestTokenPair,
  };
};

export const setProfileImageURL = (profileImageURL) => {
  return {
    type: types.SET_PROFILE_IMAGE_URL,
    profileImageURL,
  };
};

export const setProfileLinkColor = (profileLinkColor) => {
  return {
    type: types.SET_PROFILE_LINK_COLOR,
    profileLinkColor,
  };
};

export const updateWeightedStatus = (weightedStatus) => {
  return {
    type: types.UPDATE_WEIGHTED_STATUS,
    weightedStatus,
  };
};

export const setStatusImage = (statusImage) => {
  return {
    type: types.SET_STATUS_IMAGE,
    statusImage,
  };
};

export const toggleLaunchOnStartUp = (launchOnStartUp) => {
  return {
    type: types.TOGGLE_LAUNCH_ON_START_UP,
    launchOnStartUp,
  };
};

export const toggleTheme = (theme) => {
  return {
    type: types.TOGGLE_THEME,
    theme,
  };
};

export const onLogout = () => {
  return {
    type: types.ON_LOGOUT,
  };
};
