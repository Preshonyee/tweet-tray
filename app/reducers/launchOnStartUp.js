import { TOGGLE_LAUNCH_ON_START_UP, } from '../actions/actionTypes';

const launchOnStartUp = (state = false, action) => {
  switch (action.type) {
    case TOGGLE_LAUNCH_ON_START_UP:
      return action.launchOnStartUp;
    default:
      return state;
  }
};

export default launchOnStartUp;
