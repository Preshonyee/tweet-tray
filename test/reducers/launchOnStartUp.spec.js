import launchOnStartUp from '../../app/reducers/launchOnStartUp';
import { TOGGLE_LAUNCH_ON_START_UP, } from '../../app/actions/actionTypes';
import * as testData from '../testData';

describe('reducers', () => {
  describe('launchOnStartUp', () => {
    it('should handle initial state', () => {
      expect(launchOnStartUp(null, {})).toMatchSnapshot();
    });

    it('should handle TOGGLE_LAUNCH_ON_START_UP', () => {
      expect(launchOnStartUp(testData.launchOnStartUp, {
        type: TOGGLE_LAUNCH_ON_START_UP,
      })).toMatchSnapshot();
    });

    it('should handle unknown action type', () => {
      expect(launchOnStartUp(testData.launchOnStartUp, { type: 'unknown', })).toMatchSnapshot();
    });
  });
});
