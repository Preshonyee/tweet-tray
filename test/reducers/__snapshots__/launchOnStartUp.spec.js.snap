// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers launchOnStartUp should handle TOGGLE_LAUNCH_ON_START_UP 1`] = `undefined`;

exports[`reducers launchOnStartUp should handle initial state 1`] = `null`;

exports[`reducers launchOnStartUp should handle unknown action type 1`] = `true`;
